src/main/java/com/puppycrawl/tools/checkstyle/DetailNodeTreeStringPrinter.java

```diff
@@ -66,7 +64,7 @@ public static DetailNode parseJavadocAsDetailNode(DetailAST blockComment) {
         final JavadocDetailNodeParser parser = new JavadocDetailNodeParser();
         final ParseStatus status = parser.parseJavadocAsDetailNode(blockComment);
         if (status.getParseErrorMessage() != null) {
-            throw new IllegalArgumentException(getParseErrorMessage(status.getParseErrorMessage()));
+            throw new IllegalArgumentException(status.getParseErrorMessage().toString());
```

Comment:
it is not ok.
toString() method should be technical method to dump whole content to string value for logging or some technical stuff. All changes in it should be triggered by class structure changes, ...... There is good chapter in "Effective Java" book about this.
getParseErrorMessage - used for user friendly custom message, it might change to make it more user friendly. So it is responsibility of DetailNodeTreeStringPrinter to deliver message in good language to Checkstyle users.

Reply:
If you agree to make `DetailNodeTreeStringPrinter::getParseErrorMessage` public, this PR can be dropped.